python/pyspark/ml/wrapper.py

```diff
@@ -391,8 +445,52 @@ def _fit_java(self, dataset: DataFrame) -> "JavaObject":
         self._transfer_params_to_java()
         return self._java_obj.fit(dataset._jdf)
 
+    def _fit_remote(self, dataset: DataFrame) -> str:
+        """
+        Fits a remote model to the input dataset.
+
+        Examples
+        --------
+        dataset : :py:class:`pyspark.sql.DataFrame`
+            input dataset
+
+        Returns
+        -------
+        str
+            the id of the fitted remote model
+        """
+
+        def get_estimator_name(obj):
+            module = obj.__class__.__module__
+            if module is None or module == str.__class__.__module__:
+                return obj.__class__.__name__
+            else:
+                return module + '.' + obj.__class__.__name__
+
+        estimator_name = get_estimator_name(self).replace("pyspark", "org.apache.spark")
```

Comment:
You should replace it at client side. That logic could happen at servers, only when no custom plugin is given